src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -69,6 +70,11 @@ public byte[] mask(byte[] input) {
         MaskingState maskingState = new MaskingState(input, 1);
         if (AsciiCharacter.isSquareBracketOpen(maskingState.byteAtCurrentIndexMinusOne())) {
             maskingState.expandCurrentJsonPath();
+            KeyMaskingConfig keyMaskingConfig = keyMatcher.getMaskConfigIfMatched(maskingState.getMessage(), maskingState.getCurrentJsonPath());
+            if (keyMaskingConfig != null) {
+                maskArrayElementInPlace(maskingState, keyMaskingConfig);
+                trackCurrentJsonPath(maskingState);
```

Comment:
Since this is the main algorithm, it might be good to explain with comments left and right what is going on (like the rest of this file does whenever it is not obvious). This makes it easier for all of us to read the code and maintain it in the future

Reply:
I will write more docs there.